front/src/modules/simulationResult/components/TrainDetailsV2.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useMemo, useState } from 'react';
+
+import { useTranslation } from 'react-i18next';
+
+import type { TrainSpaceTimeData } from 'applications/operationalStudies/types';
+import { convertDepartureTimeIntoSec } from 'applications/operationalStudies/utils';
+import type { PositionsSpeedTimes } from 'reducers/osrdsimulation/types';
+
+import { useChartSynchronizerV2 } from './ChartHelpers/ChartSynchronizerV2';
+
+/**
+ * Given the routeAspects and a timePosition, returns the bounds [start, end]
+ * of the currently occupied canton (in meters)
+ *
+ * @param routeAspects
+ * @param time
+ * @returns current Canton occupied by current train, [start, end] in meters
+ * @TODO do not work with colors as string as soon as possible
+ */
+const getOccupancyBounds = (
+  routeAspects: TrainSpaceTimeData['signal_updates'],
+  timePosition: Date
+): [number, number] => {
+  const relevantAspect = routeAspects.find((routeAspect) => {
+    const relevantStartTime = routeAspect.time_start || new Date();
+    const relevantEndTime = routeAspect.time_end || new Date();
+
+    return (
+      relevantStartTime < timePosition &&
+      relevantEndTime >= timePosition &&
+      routeAspect.color === -65536 // Equivalent to 'rgba(255, 0, 0, 255)' in bits
+    );
+  });
+  return relevantAspect ? [relevantAspect.position_start, relevantAspect.position_end] : [0, 0];
+};
+
+type TrainDetailsV2Props = {
+  projectedTrain: TrainSpaceTimeData;
+  selectedTrainDepartureTime: string;
+};
+
+const TrainDetailsV2 = ({ projectedTrain, selectedTrainDepartureTime }: TrainDetailsV2Props) => {
+  const [{ headPosition, tailPosition, speed }, setLocalPositionValues] = useState<
+    PositionsSpeedTimes<Date>
+  >({} as PositionsSpeedTimes<Date>);
+
+  useChartSynchronizerV2(
+    (_, positionValues) => {
+      setLocalPositionValues(positionValues);
+    },
+    'train-detail',
+    []
+  );
+
+  const { t } = useTranslation(['simulation']);
+
+  const formattedSignalUpdates: TrainSpaceTimeData['signal_updates'] = useMemo(() => {
+    const convertedDepartureTime = `1900-01-01${selectedTrainDepartureTime.substring(10)}`;
```

Comment:
Why don't use sec2d3datetime here ?

Reply:
It changed the logic a bit, this component is fixed now :)